client/lib/preferences-helper/style.scss

```diff
@@ -1,61 +1,139 @@
+	}
+
+	tr:last-child td {
+		border-bottom: none;
 	}
 }
 
-.preferences-helper__unset {
-	color: var( --color-accent );
-	cursor: pointer;
+.preferences-helper__header-unset {
+	text-align: center;
 }
 
-.preferences-helper__preference-header {
-	text-transform: initial;
-	margin-top: 5px;
+.preferences-helper__header-name {
+	text-align: right;
+}
+
+.preferences-helper__no-preferences {
+	text-align: center;
+
+	font-size: 1.5rem;
 	font-weight: 600;
 	white-space: nowrap;
 	overflow: hidden;
+}
 
-	button {
-		display: inline;
-		margin-right: 5px;
+.preferences-helper__preferences-table .boolean-preference input {
+	margin: initial;
+}
+
+.preferences-helper__preferences-table .array-preference,
+.preferences-helper__preferences-table .object-preference {
+	margin: initial;
+	list-style: none;
+
+	li {
+		font-weight: normal;
 	}
 }
 
-.preferences-helper__save-pref-button {
+.preferences-helper__preferences-table .object-preference li {
+	margin-bottom: 8px;
+}
+
+.preferences-helper__preferences-table .object-preference__property {
+	display: block;
+	margin-bottom: 4px;
+	font-family: monospace;
+}
+
+.preferences-helper__preferences-table .array-preference {
+	font-family: monospace;
+}
+
+.preferences-helper__preferences-table .preference__unset {
+	text-align: center;
 	color: var( --color-accent );
+
+	button {
+		cursor: pointer;
+	}
+}
+
+.preferences-helper__preferences-table .preference__name {
+	text-align: right;
+	font-family: monospace;
+	font-weight: 600;
+}
+
+.preferences-helper__preferences-table .boolean-preference__save,
+.preferences-helper__preferences-table .number-preference__save,
+.preferences-helper__preferences-table .string-preference__save {
+	margin: 0 4px;
+	padding: 4px 8px;
+
+	border-radius: 2px;
+	background-color: var( --color-accent );
+	color: var( --color-text-inverted );
+	font-weight: 600;
+
 	cursor: pointer;
 	&:disabled {
-		color: var( --color-neutral );
+		background-color: var( --color-neutral );
 	}
 }
 
-.preferences-helper__reset-pref-button {
+.preferences-helper__preferences-table .boolean-preference__reset,
+.preferences-helper__preferences-table .number-preference__reset,
+.preferences-helper__preferences-table .string-preference__reset {
+	margin: 0 4px;
+	padding: 4px 8px;
+
+	border-radius: 2px;
+	border: 1px solid var( --color-warning );
 	color: var( --color-warning );
+	font-weight: 600;
+
 	cursor: pointer;
+
 	&:disabled {
-		color: var( --color-neutral );
+		color: var( --color-text-subtle );
 	}
 }
 
-.boolean-preference {
-	input {
-		appearance: checkbox;
-	}
+.preferences-helper__preferences-table .number-preference input,
+.preferences-helper__preferences-table .string-preference input {
+	margin: 4px 1rem 4px 0;
+}
+
+.preferences-helper__preferences-table .boolean-preference input {
+	margin: 8px 1rem 8px 0;
+	appearance: checkbox;
 }
 
 .environment.is-prefs:hover .preferences-helper__current-preferences.card {
 	display: block;
+	max-width: 40vw;
```

Comment:
This assumes that the person navigating the preferences is using a large screen, and I'm not sure it will always be the case - here's how it looks on a 1360px width for example:

![](https://cldup.com/-t0FZT8Iql.png)

Actually, there is a horizontal scroll inside the preferences helper when viewing it on my 24-inch monitor on the full screen, and I'd say that's undesirable.